File: src/dotnet/Trivi/Trivi.WebGui/wwwroot/js/ts/custom/domain/helpers/new-questions/multiple-choice.ts
import { QuestionType } from "../../enums/question-type";
import { PutMultipleChoiceApiRequest, PutQuestionApiRequest } from "../../models/question-models";
import { NewQuestion } from "./new-question";


export class NewMultipleChoice extends NewQuestion<PutMultipleChoiceApiRequest>
{

    public getQuestionType = () => QuestionType.MultipleChoice;
    protected readonly _defaultPrompt = "New multiple choice";

    public toForm(): PutQuestionApiRequest
    {
        const result: PutMultipleChoiceApiRequest = {
            collectionId: this._collectionId,
            prompt: this._defaultPrompt,
        };

        return result;
    }

}

File: src/dotnet/Trivi/Trivi.Lib/Domain/Models/MultipleChoice.cs
using Trivi.Lib.Domain.Forms;
using Trivi.Lib.Domain.Other;

namespace Trivi.Lib.Domain.Models;

public class MultipleChoice : Question
{
    public override QuestionType QuestionType => QuestionType.MultipleChoice;

    public static MultipleChoice FromRequestForm(QuestionId questionId, MultipleChoiceForm form)
    {
        return new()
        {
            Id = questionId,
            CollectionId = form.CollectionId,
            Prompt = form.Prompt,
        };
    }
}
